common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -2452,20 +2462,20 @@ def test_course_deleted_signal(self, default):
             with self.store.default_store(default):
                 self.assertIsNotNone(self.store.thread_cache.default_store.signal_handler)
 
-                with mock_signal_receiver(SignalHandler.course_deleted) as receiver:
-                    self.assertEqual(receiver.call_count, 0)
+                signal_handler.send.assert_not_called()
 
-                    # Create a course
-                    course = self.store.create_course('org_x', 'course_y', 'run_z', self.user_id)
-                    course_key = course.id
+                # Create a course
+                course = self.store.create_course('org_x', 'course_y', 'run_z', self.user_id)
+                course_key = course.id
 
-                    # Delete the course
-                    course = self.store.delete_course(course_key, self.user_id)
+                # Delete the course
+                course = self.store.delete_course(course_key, self.user_id)
 
-                    # Verify that the signal was emitted
-                    self.assertEqual(receiver.call_count, 1)
+                # Verify that the signal was emitted
+                signal_handler.send.assert_called_with('course_deleted', course_key=course_key)
 
     @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    @expectedFailure  # We know that split and mongo create orphans in this case
```

Comment:
@adampalay: Split and old-mongo behave the same way, and create orphans when you delete the child of a DIRECT_ONLY_CATEGORY which itself has children. The child (vertical) is deleted, but the children of that are not deleted.